consensus/polybft/stake/stake_manager.go

```diff
@@ -230,20 +169,66 @@ func (s *stakeManager) updateWithReceipts(
 	return nil
 }
 
+// Get Active Validator from StakeManager smart contract
+func (s *stakeManager) getActiveValidators() (validator.ValidatorStakeMap, error) {
+	provider, err := s.blockchain.GetStateProviderForBlock(s.blockchain.CurrentHeader())
+	if err != nil {
+		return nil, err
+	}
+
+	stakeManagerContract := contract.NewContract(
+		ethgo.Address(s.stakeManagerContractAddr),
+		contractsapi.StakeManager.Abi,
+		contract.WithProvider(provider),
+	)
+
+	res, err := stakeManagerContract.Call("getActiveValidators", ethgo.Latest)
+	if err != nil {
+		return nil, fmt.Errorf("failed to call getActiveValidators: %w", err)
+	}
+
+	type ActiveValidator struct {
+		Address types.Address `json:"addr" mapstructure:"addr"`
+		BlsKey  [4]*big.Int   `json:"blsKey" mapstructure:"blsKey"`
+		Stake   *big.Int      `json:"stake" mapstructure:"stake"`
+	}
+
+	var decodedValidators []ActiveValidator
+
+	if err = mapstructure.Decode(res["0"], &decodedValidators); err != nil {
+		return nil, fmt.Errorf("failed to decode getActiveValidators response: %w", err)
+	}
+
+	validatorSet := make(validator.ValidatorStakeMap, len(decodedValidators))
+
+	for _, v := range decodedValidators {
+		publicKey, err := bls.UnmarshalPublicKeyFromBigInt(v.BlsKey)
+		if err != nil {
+			return nil, fmt.Errorf("failed to unmarshal BLS public key: %w", err)
+		}
+
+		validatorSet[v.Address] = &validator.ValidatorMetadata{
+			Address:     v.Address,
+			VotingPower: v.Stake,
+			BlsKey:      publicKey,
+			IsActive:    true,
+		}
+	}
+
+	return validatorSet, nil
+}
+
 // UpdateValidatorSet returns an updated validator set
 // based on stake change (transfer) events from ValidatorSet contract
 func (s *stakeManager) UpdateValidatorSet(epoch uint64, maxValidatorSetSize uint64,
 	oldValidatorSet validator.AccountSet) (*validator.ValidatorSetDelta, error) {
 	s.logger.Info("Calculating validators set update...", "epoch", epoch)
 
-	fullValidatorSet, err := s.state.getFullValidatorSet(nil)
+	stakeMap, err := s.getActiveValidators()
```

Comment:
Getting only active validators is not enough. Imagine you have 5 active validators and 2 validators that registered and staked in the epoch that is closing. You need to consider them as well.

This might be updated in some other part of the code, but stake manager was the one that was keeping the entire history of all validators and their current stake, and it was also responsible to updating that full validator set with stake events. That was done in that updateWithReceipts function.